network-api/networkapi/wagtailpages/pagemodels/campaigns.py

```diff
@@ -422,11 +436,30 @@ class BanneredCampaignPage(PrimaryPage):
 
     show_in_menus_default = True
 
+    def get_intro(self):
+        if self.use_intro_from_parent:
+            parent = self.get_parent()
+            if parent.specific.intro:
+                return parent.specific.intro
+            else:
+                return ''
+        else:
+            return self.intro
+
     def get_context(self, request):
         context = super().get_context(request)
         context['related_posts'] = get_content_related_by_tag(self)
-        return get_page_tree_information(self, context)
+        context = get_page_tree_information(self, context)
+
+        return context
 
     class Meta:
         verbose_name = "Banner Page"
         verbose_name_plural = "Banner pages"
+
+    def clean(self):
+        if self.use_intro_from_parent and self.intro:
+            raise ValidationError({
+                'use_intro_from_parent': ValidationError("Please use only one intro option."),
+                'intro': ValidationError("Please use only one intro option.")
+            })
```

Comment:
Nice. A different check than I meant, but a good one still!


I was thinking about something like (pseudo-code):
```
if use_intro_from_parent:
    parent = get_parent
    if not has_intro_field(parent):
        raise ValidationError
```

Reply:
Ahh yes thanks for this. I've added validation for this as well now.